### PR TITLE
Post api/v1/services

### PR DESCRIPTION
**Pull request by Jehad91**
Relates #61
image

### CODE REVIEW

File: server/controllers/services/postService.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+const postService: RequestHandler = async (req, res, next) => {
+  const {
+    name, description, price, image,
+  } = req.body;
+  const categoryId = req.body.category_id;
+
+  try {
+    await servicesSchema.validate({
+      name, description, price, image, categoryId,
+    }, { abortEarly: false });
```
Comment (Mu7ammadAbed) [inline server/controllers/services/postService.ts:15]:
```suggestion
  try {
    const {
      name, description, price, image, categoryId,
    } = await servicesSchema.validate(req.body, { abortEarly: false });
```

File: test/index.test.ts [hunk 2]
```diff
@@ -198,3 +198,52 @@ describe('Test to login', () => {
...
+describe('Test post service', () => {
+  it('should return status 201', async () => {
+    const res = await supertest(app)
+      .post('/api/v1/services')
+      .send({
+        name: 'test',
+        description: 'test',
+        price: 10,
+        image: 'ka;hkjdgh;askdh',
+        category_id: 1,
```
Comment (Mu7ammadAbed) [inline test/index.test.ts:211]:
```suggestion
        categoryId: 1,
```

File: test/index.test.ts [hunk 3]
```diff
@@ -198,3 +198,52 @@ describe('Test to login', () => {
...
+describe('Test post service', () => {
+  it('should return status 201 and text successfuly message', async () => {
+    const res = await supertest(app)
+      .post('/api/v1/services')
+      .send({
+        name: 'test',
+        description: 'test',
+        price: 10,
+        image: 'ka;hkjdgh;askdh',
+        category_id: 1,
```
Comment (Mu7ammadAbed) [inline test/index.test.ts:227]:
```suggestion
        categoryId: 1,
```

File: test/index.test.ts [hunk 4]
```diff
@@ -198,3 +198,52 @@ describe('Test to login', () => {
...
+describe('Test post service', () => {
+  it('should return status code 400 and error message for price validation', async () => {
+    const res = await supertest(app)
+      .post('/api/v1/services')
+      .send({
+        name: 'test',
+        description: 'test',
+        price: 0,
+        image: 'ka;hkjdgh;askdh',
+        category_id: 1,
```
Comment (Mu7ammadAbed) [inline test/index.test.ts:243]:
```suggestion
        categoryId: 1,
```

File: server/queries/services/postServiceQuery.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+const postServiceQuery = (
+  name: String,
+  description: String,
+  price: number,
+  image: String,
+  categoryId: number,
+) => {
+  const sql = {
+    text: 'INSERT INTO services(name, description, price, image, category_id) VALUES ($1,$2,$3,$4,$5) RETURNING *',
```
Comment (Mu7ammadAbed) [inline server/queries/services/postServiceQuery.ts:11]: If you are not using the returned value then why would you return it in the first place?
```suggestion
    text: 'INSERT INTO services(name, description, price, image, category_id) VALUES ($1,$2,$3,$4,$5);',
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by MohammadAlHabil — APPROVED** [review]

**Review by IsraaHamdi — APPROVED** [review]

**Review by MostafaQanoo — APPROVED** [review]

**Pull request merged**